kafka/client_async.py

```diff
@@ -499,14 +499,15 @@ def is_ready(self, node_id, metadata_priority=True):
         return True
 
     def _can_send_request(self, node_id):
-        with self._lock:
-            if node_id not in self._conns:
-                return False
-            conn = self._conns[node_id]
-            return conn.connected() and conn.can_send_more()
+        conn = self._conns.get(node_id)
+        if not conn:
+            return False
+        return conn.connected() and conn.can_send_more()
 
     def send(self, node_id, request):
-        """Send a request to a specific node.
+        """Send a request to a specific node. Bytes are placed on an
+        internal per-connection send-queue. Actual network I/O will be
+        triggered in a subsequent call to .poll()
```

Comment:
Does this create any new scenarios where we now need to make sure to call `poll()`?
In other words, will this break any behavior that used to work fine w/o ever calling `poll()`?

I can't think of any--looks like metadata refresh picks this up automatically since it relies on `maybe_connect`, but thought it's a question still worth thinking through...

Reply:
It is possible, yes, but they should be rare. The main culprits would be blocking loops attempting to connect without calling poll(), or not calling poll() unless there are in-flight-requests. I think I've found and fixed those issues, but definitely keep an eye open for others.